TM1py/Services/CubeService.py

```diff
@@ -249,6 +250,39 @@ def search_for_dimension_substring(self, substring: str, skip_control_cubes: boo
         cube_dict = {entry['Name']: [dim['Name'] for dim in entry['Dimensions']] for entry in response.json()['value']}
         return cube_dict
 
+    def enable_cube_rule(self, cube: Cube) -> None:
+        """
+        Enable a cube rule from its base64-encoded hash if it exists.
+
+        :param cube: An instance of a Cube.
+        """
+        current_rule = cube.rules.text
+        if not current_rule:
+            # If there is no rule, there is nothing to do.
+            return
+        try:
+            current_rule = current_rule[1:] if current_rule.startswith('#') else current_rule
+            cube.rules = base64.b64decode(current_rule).decode('utf-8')
+        except Exception:
+            raise ValueError(f"Current rule is not decodable by b64decode standards")
+
+        self.update(cube)
+
+    def disable_cube_rule(self, cube: Cube) -> None:
+        """
+        Disable a cube rule by saving its base64-encoded hash and commenting each line.
+
+        :param cube: An instance of a Cube.
+        """
+        current_rule = cube.rules.text
+        if not current_rule:
+            # If there is no rule, there is nothing to do.
+            return
+
+        # Save the current rule as a base64-encoded hash
+        cube.rules = f"#{base64.b64encode(current_rule.encode('utf-8')).decode('utf-8')}"
```

Comment:
Maybe prefix the actual encoding in the comment with something like `b64 encoded rule=` for transparency purposes

Reply:
fixed in commit f6ee7209d4b830bc4a0e8b733508bc27bc9f4490